modules/cudf/src/data_frame.ts

```diff
@@ -1587,4 +1588,76 @@ export class DataFrame<T extends TypeMap = any> {
       this.names.reduce((cols, name) => ({...cols, [name]: this.get(name).isNotNull()}),
                         {} as SeriesMap<{[P in keyof T]: Bool8}>));
   }
+
+  /**
+   * Replace null values with a scalar value.
+   *
+   * @param value The scalar value to use in place of nulls.
+   * @param memoryResource The optional MemoryResource used to allocate the result Column's device
+   *   memory.
+   *
+   * @example
+   * ```typescript
+   * import {DataFrame, Series} from '@rapidsai/cudf';
+   *
+   * const df = new DataFrame({
+   *  a: Series.new([0, null, 2]);
+   *  b: Series.new([null, null, null]);
+   * });
+   *
+   * df.replaceNulls(1);
+   * // return {
+   * //    a: [0, 1, 2],
+   * //    b: [1, 1, 1],
+   * // }
+   * ```
+   */
+  replaceNulls(value: T['scalarType']|any, memoryResource?: MemoryResource): DataFrame;
```

Comment:
The TypeMap T doesn't have a `scalarType`, so this overload is invalid.

Reply:
Any suggestions on replacements? is `any` fine?

or would `number|string|boolean` cover all the scalar types.

Reply:
Nevermind... I think our chats about datatypes helped out.

```
replaceNulls(value: (DataType&keyof T)['scalarType'],
               memoryResource?: MemoryResource): DataFrame<T>;
```

Since the key of T is a DataType, we should be able to just access the `scalarType`.